### PR TITLE
UniProt GOA Added parser for GPI version 1.2

### PR DESCRIPTION
**Pull request by svalqui**
- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``pre-commit`` locally,
and understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

Adds a parser for the latest version of UniProt Go Annotations, Gene Product Information, version 1.2

### CODE REVIEW

**Comment by peterjc** [comment]
@idoerg are you able to give this GOA/GPI parser work a quick look over?

File: NEWS.rst [hunk 1]
```diff
@@ -27,7 +27,8 @@ New module Bio.PDB.SASA implements the Shrake-Rupley algorithm to calculate
 atomic solvent accessible areas without third-party tools.
 
 Bio.UniProt.GOA now parses Gene Product Information (GPI) files version 1.2,
-files can be found on the EBI ftp site ftp://ftp.ebi.ac.uk/pub/databases/GO/goa/
+files can be downloaded from the EBI ftp site:
+ftp://ftp.ebi.ac.uk/pub/databases/GO/goa/
```
Comment (peterjc) [inline NEWS.rst:31]: If you wanted two lines, how about:

```
Bio.UniProt.GOA now parses Gene Product Information (GPI) version 1.2, files
can be found on the EBI ftp site ftp://ftp.ebi.ac.uk/pub/databases/GO/goa/
```

Comment (FriedbergLab) [reply NEWS.rst:31]: @peterjc just a rewording of the documentation? Sure.

Comment (idoerg) [reply NEWS.rst:31]: @peterjc Just a rewording of the documentation? Sure.

Comment (peterjc) [reply NEWS.rst:31]: I meant could you look at the entire pull request, adding v1.2 GPI parsing.

Comment (idoerg) [reply NEWS.rst:31]: Looks good to me. Can approve it now. If you want, I can test it, but won't be before the weekend.

Comment (peterjc) [reply NEWS.rst:31]: Testing would be great, even if a week away. Thanks!

Comment (svalqui) [reply NEWS.rst]: @peterjc , thank you for having a look at this, I'm having an issue with the pre-commit hooks, the 2 lines you suggested fails doc8, line too long, my git pre-commit is checking flake8 and black but doc8, that is why the initial commit failed as well, I'll try rolling it back.

@idoerg , thank you.

Cheers,

Comment (svalqui) [reply NEWS.rst:31]: @peterjc , git reset didn't work :( had to do another commit, if you could squash them it would look better, thank you.

Comment (peterjc) [reply NEWS.rst:31]: I must have miscounted, I thought that version would fit fine. Sorry.

Anyway, there's no problem doing a git merge here - everything is authored by you.

**Comment by peterjc** [comment]
Thank you - apologies, we could have merged this months ago.

**Pull request merged**